### PR TITLE
DOC: Update documentation of np.clip

### LINKED ISSUE #15061: np.clip: change in precedence between minimum and maximum when maximum < minimum
Greetings!

First, thanks to the NumPy community for your efforts.

Second, In NumPy 1.16.5 I observe the following:

```python
>>> import numpy as np
>>> np.clip(2, 3, 0)
3
```

In NumPy 1.17.4, I get a different result:

```python
>>> import numpy as np
>>> np.clip(2, 3, 0)
0
```
I see that the difference has to do with which constraint gets precedence. Was this an intentional change in precedence then?

### Reproducing code example:

```python
import numpy as np
assert np.clip(2, 3, 0), 3
```

### Numpy/Python version information:

```python
>>> import sys, numpy; print(numpy.__version__, sys.version)
1.17.4 3.7.4 (default, Dec  3 2019, 11:37:16) 
[GCC 5.4.0 20160609]
```

### PR DESCRIPTION
**Pull request by bharatr21**
Fixes #15061 by informing users and  updating the documentation of np.clip

### CODE REVIEW

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -2033,7 +2033,8 @@ def clip(a, a_min, a_max, out=None, **kwargs):
     is specified, values smaller than 0 become 0, and values larger
     than 1 become 1.
 
-    Equivalent to but faster than ``np.maximum(a_min, np.minimum(a, a_max))``.
+    Equivalent to but faster than ``np.maximum(a_min, np.minimum(a, a_max))``
+    only if ``a_min < a_max`` else returns ``a_max``.
```
Comment (tylerjereddy) [inline numpy/core/fromnumeric.py]: I'd be surprised if it were this straightforward, but happy to be wrong :)

I though it was more like if `amin < amax`, strange things can happen depending on the nature of the input.

Comment (mattip) [reply numpy/core/fromnumeric.py]: Maybe simply "Assumes `a_min < a_max`"

Comment (eric-wieser) [reply numpy/core/fromnumeric.py]: I think it actually _is_ this straightforward (whereas it didn't use to be). Bring deliberately vague here is probably good anyway though.

**Comment by mattip** [comment]
Thanks @Bharat123rox

**Pull request merged**